UndertaleModLib/Util/TextureWorker.cs

```diff
@@ -1,217 +1,250 @@
+                return image.Clone();
             }
 
-            return bm;
+            // Clone and resize using bilinear interpolation
+            IMagickImage<byte> newImage = image.Clone();
+            newImage.InterpolativeResize(width, height, PixelInterpolateMethod.Bilinear);
+            return newImage;
         }
 
-        // This should perform a high quality resize.
-        // Grabbed from https://stackoverflow.com/questions/1922040/how-to-resize-an-image-c-sharp
-        public static Bitmap ResizeImage(Image image, int width, int height)
+        /// <summary>
+        /// Reads collision mask data from the given file path, and required width/height.
+        /// </summary>
+        /// <exception cref="Exception">If the loaded image dimensions do not match the required width/height</exception>
+        public static byte[] ReadMaskData(string filePath, int requiredWidth, int requiredHeight)
         {
-            var destRect = new Rectangle(0, 0, width, height);
-            var destImage = new Bitmap(width, height);
-
-            destImage.SetResolution(image.HorizontalResolution, image.VerticalResolution);
-
-            using (var graphics = Graphics.FromImage(destImage))
+            List<byte> bytes;
+            using (MagickImage image = ReadBGRAImageFromFile(filePath))
             {
-                graphics.CompositingMode = CompositingMode.SourceCopy;
-                graphics.CompositingQuality = CompositingQuality.HighQuality;
-                graphics.InterpolationMode = InterpolationMode.HighQualityBicubic;
-                graphics.SmoothingMode = SmoothingMode.HighQuality;
-                graphics.PixelOffsetMode = PixelOffsetMode.HighQuality;
-
-                using (var wrapMode = new ImageAttributes())
+                // Verify width/height match required width/height
+                if (image.Width != requiredWidth || image.Height != requiredHeight)
                 {
-                    wrapMode.SetWrapMode(WrapMode.TileFlipXY);
-                    graphics.DrawImage(image, destRect, 0, 0, image.Width, image.Height, GraphicsUnit.Pixel, wrapMode);
+                    throw new Exception($"{filePath} is not the proper size to be imported! The proper dimensions are width: {requiredWidth} px, height: {requiredHeight} px.");
                 }
-            }
 
-            return destImage;
-        }
+                // Get image pixels, and allocate enough capacity for mask
+                IPixelCollection<byte> pixels = image.GetPixels();
+                bytes = new((requiredWidth + 7) / 8 * requiredHeight);
 
-        public static byte[] ReadMaskData(string filePath)
-        {
-            Bitmap image = ReadImageFromFile(filePath);
-            List<byte> bytes = new List<byte>();
+                // Get white color, used to represent bits that are set
+                IMagickColor<byte> white = MagickColor.FromRgba(255, 255, 255, 255);
```

Comment:
surely imagemagick has a "white" constant?

Reply:
Good catch, I just dug around the docs for a bit and finally found the color constants.